### PR TITLE
chore: validate InfluxDB query bug fix and update one more otel reference

### PR DESCRIPTION
**Pull request by jacobmarble**
Confirmed this bug is fixed, and made a few small changes.

https://github.com/influxdata/influxdb_iox/issues/7225

### CODE REVIEW

File: jaeger-influxdb/internal/queries.go [hunk 1]
```diff
@@ -18,20 +18,12 @@ func traceIDToString(traceID model.TraceID) string {
...
-		whereClause = "false"
+	traceIDStrings := make([]string, len(traceIDs))
+	for i, traceID := range traceIDs {
+		traceIDStrings[i] = traceIDToString(traceID)
 	}
-	query := fmt.Sprintf("SELECT * FROM %s WHERE %s",
-		table, whereClause)
-	return query
+	return fmt.Sprintf(`SELECT * FROM %s WHERE "%s" IN ('%s')`,
+		table, common.AttributeTraceID, strings.Join(traceIDStrings, `','`))
```
Comment (philjb) [inline jaeger-influxdb/internal/queries.go:26]: presumably traceIDs won't be zero anymore (otherwise the in RHS will be empty.

**Review by philjb — APPROVED** [review]
i see a change for a new error and a change on traceid queries that could be a fix or a bug - presumably the former!

File: jaeger-influxdb/internal/queries.go (hunk 1 above)
Comment (jacobmarble) [reply jaeger-influxdb/internal/queries.go:26]: The traceIDs may be empty. If so, the where clause is `WHERE "trace_id" IN ('')`, which is valid, and returns zero results.

Of course, this assumes the trace_id column contains no empty string values, which I cannot guarantee.

Comment (philjb) [reply jaeger-influxdb/internal/queries.go:26]: ah for string/char columns that works. I tested on an int column. 🤦

**Pull request merged**